### PR TITLE
Robert Round Robin 

### PR DESCRIPTION
**Pull request by allaway**
### Purpose

Partially addresses #146.

### Directions for reviewers

Please make sure I didn't negatively impact anything.  

#### Which areas should receive a particularly close look?

In particular, please make sure the TODOs that I deleted were addressed.
Please also check the sections where I heavily edited and rephrased stuff to make sure I didn't obfuscate the message. 

#### Is there anything that you want to discuss further?

I added one TODO for close evaluation.

#### Is the pull request ready for review?

Yes

### Manuscript checklist

Unless otherwise noted above, this PR will be considered ready for review when all four items have been checked.

- [x] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#manubot-markdown)]
- [x] All citations follow the [Manubot citation instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#citations)
- [ ] All changes or additions to tables follow the [Manubot table instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#tables)
- [ ] All figures follow the [Manubot figure instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#figures)
- [x] All terms that need to be added to glossary have been added to #50 
- [x] There are no new spelling errors identified by the Manubot spellchecker

### CODE REVIEW

**Comment by allaway** [comment]
I think @cgreene is next in line for a round robin review. As per last week's discussion I tagged him for review but also tagged @jaclyn-taroni and @jaybee84 in case you all want to review my changes.

File: content/01.synopsis.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 Machine learning is often a perfect fit for extracting disease-relevant patterns from these high dimensional datasets.
 Often, machine learning methods require many samples to identify reproducible and biologically meaningful patterns.
 With rare diseases, biological specimens, and consequently data, are limited due to the rarity of the condition.
 In this perspective, we outline the challenges and emerging solutions for using machine learning in rare disease settings.
-We aim to spur the development of powerful machine learning techniques for rare diseases.
-We also note that precision medicine presents a similar challenge, in which a common disease is partitioned into small subsets of patients with shared etiologies and treatment strategies.
-Advances from rare disease research are likely to be highly informative for other applications as well.
+We also note that precision medicine presents a similar challenge, in which a common disease is partitioned into small subsets of patients with shared etiologies and treatment strategies. [TODO: Are we still talking about precision medicine? I thought we decided against that?]
```
Comment (jaclyn-taroni) [inline content/01.synopsis.md]: To me, this seems worth stating here – even if you work on a common disease that has many subtypes, some of which are rare, this perspective is for you!

File: content/03.heterogeneity.md [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+To ensure that the learned representations are generalizable to other data, the features learned by the model can be constrained through methods like regularization [@doi:10.1371/journal.pgen.1004754, @doi:10.1002/sim.6782]. 
+Representation learning generally requires many samples when applied to complex biological systems and therefore may appear to aggravate the curse of dimensionality. 
 However, it can be a powerful tool to learn low-dimensional patterns from large datasets and then find those patterns in smaller, related datasets. 
-In the later sections of this perspective, we will discuss this method of leveraging large datasets to reduce dimensionality in smaller datasets, also known as feature-representation-transfer. 
+In later sections, we will discuss this method of leveraging large datasets to reduce dimensionality in smaller datasets, also known as feature-representation-transfer learning. 
 
 ![Dimension reduction can help manage the curse of dimensionality in rare disease data](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/content/images/figures/pdfs/dimensionality-reduction.pdf){#fig:1}
```
Comment (jaclyn-taroni) [inline content/03.heterogeneity.md]: Do you want to expand this figure legend a bit? Is it worth adding lettering to this figure so you can cite it in the text at multiple points?

File: content/04.model-complexity.md [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
-A cascade learning approach for identifying rare disease patients from electronic health records from the general population utilized independent steps for feature extraction (using natural language processing based word2vec [@https://arxiv.org/abs/1301.3781v343]), preliminary prediction using an ensemble of decision trees, and prediction refinement using similarity of data points to resolve sample labels [@pmid:30815073]. 
+Alternatively, resampling with replacement ('bootstrapping') helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).
```
Comment (jaclyn-taroni) [inline content/04.model-complexity.md]:
```suggestion
Alternatively, resampling with replacement ("bootstrapping") helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).
```

File: content/04.model-complexity.md [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
+Ensemble learning methods like random forests use _bagging_ (bootstrap aggregation) of independent decision trees that use similar parameters but different paths to form a consensus about the important predictive features [@https://doi.org/10.1023/A:1010933404324; @doi:10.1186/1472-6947-13-134; @doi:10.1214/aos/1031689014; @doi:10.1177/2045894019890549; @doi:10/btzfh6].
```
Comment (jaclyn-taroni) [inline content/04.model-complexity.md:10]: Why does bootstrapping get quotes but bootstrap aggregation does not? Same question about ensemble learning.

File: content/04.model-complexity.md [hunk 5]
```diff
@@ -1,32 +1,33 @@
...
-On the other hand, in immune cell signature discovery, elastic-net regression has been used to reduce the feature space and was found to outperform other regression approaches [@doi:10.1111/j.1467-9868.2005.00503.x; @doi:10.1016/j.compbiomed.2015.10.008; @doi:10.1186/1471-2105-14-198; @doi:10.1186/s12859-019-2994-z]. 
-Regularization methods like LASSO or elastic-net have been methods of choice for making models simpler by reducing the feature space; these methods should be explored while working with rare disease datasets. 
+Rare variant discovery and immune cell signature discovery studies, like rare diseases, face challenges of the sparsity of observations, and may be a useful model for examining the utility of regularization in scenarios with limited signal.
```
Comment (jaclyn-taroni) [inline content/04.model-complexity.md]:
```suggestion
Rare variant discovery and immune cell signature discovery studies, like rare diseases, face challenges of the sparsity of observations, and may be useful models for examining the utility of regularization in scenarios with limited signal.
```

File: content/07.outlook.md [hunk 6]
```diff
@@ -1,28 +1,27 @@
...
+While there are many techniques to collate rare data from different sources, low-quality data may hurt the end goal even if it adds to the size of the dataset.
+In our experience, collaboration with domain experts has proved to be critical in gaining insight into potential sources of variation in the datasets.
+An anecdotal example from the authors' personal experience: conversations with a rare disease clinician revealed that samples in a particular tumor dataset were collected using vastly different surgical techniques (laser ablation and excision vs standard excision). 
+This information that was not readily available to non-experts, but was obvious the clinician. 
+Such instances underline the fact that continuous collaboration with domain experts is needed to generate robust datasets in the future.[TODO: I stripped a lot of details from this anecdote, because I would like the participants (other than us) to remain relatively anonymous, and I am not trying to appears as if I am criticizing people who are sharing data if it is not perfect. The todo is - at the current level of detail, is this still worth including?]
```
Comment (jaclyn-taroni) [inline content/07.outlook.md]: I think so.

**Review by jaclyn-taroni — APPROVED** [review]
I think this looks good! The most substantive comment I left was about fleshing out the Figure 1 legend. I don't need to re-review since I expect I will have the opportunity to take another look when this goes from Google Doc back to Manubot. I am happy to take another look if you'd like, though!

File: content/01.synopsis.md (hunk 1 above)
Comment (allaway) [inline content/01.synopsis.md]:
```suggestion
We also note that precision medicine presents a similar challenge, in which a common disease is partitioned into small subsets of patients with shared etiologies and treatment strategies. 
```

File: content/04.model-complexity.md (hunk 3 above)
Comment (allaway) [inline content/04.model-complexity.md]:
```suggestion
Alternatively, resampling with replacement (bootstrapping) helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).
```

File: content/04.model-complexity.md (hunk 4 above)
Comment (allaway) [reply content/04.model-complexity.md:10]: Upon reflection I think I prefer it with no quotes, so I changed this for bootstrapping. Maybe need to scrub the rest of the paper for this too....

File: content/07.outlook.md (hunk 6 above)
Comment (allaway) [inline content/07.outlook.md]:
```suggestion
Such instances underline the fact that continuous collaboration with domain experts is needed to generate robust datasets in the future.
```

File: content/03.heterogeneity.md (hunk 2 above)
Comment (allaway) [inline content/03.heterogeneity.md]:
```suggestion
![Dimension reduction can help manage the curse of dimensionality in rare disease data. A) Multiple datasets (shapes) with multiple phenotypes (purple, green) are combined for an analysis. The data (e.g., transcriptomic data) are highly dimensional, having thousands of features (f1-f100000). B) Evaluating the features, it appears that a combination of features (e.g., expressed genes) partition the purple samples from the green samples. C) Applying a dimensionality reduction method (e.g., PCA) condenses these features into new features (e.g., New Feature 1, a combination of f1, f2 .... f100000, and New Feature 2, a different combination of f1, f2 .... f100000). New Feature 1 describes the difference in input dataset (shapes) while New Feature 2 describes the difference in phenotype (color). D) New features (F1-F1000) can be used to interrogate the biology of the input samples, develop classification models, or use other analytical techniques that would have been more difficult with the original dataset dimensions.](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/content/images/figures/pdfs/dimensionality-reduction.pdf){#fig:1}
```

**Pull request merged**